entities/entities/nut_item.lua

```diff
@@ -51,14 +51,22 @@ if (SERVER) then
 				and "models/props_junk/cardboard_box004a.mdl"
 				or itemTable.worldModel
 		end
-
-		self:SetSkin(itemTable.skin or 0)
+		self:SetModel("models/props_junk/watermelon01.mdl")
 		self:SetModel(model)
+		self:SetSkin(itemTable.skin or 0)
+
+		if itemTable.groups then -- this has to be done after the model is set, hence why it looks a little messy
+			for k, v in pairs(itemTable.groups) do
+				if isstring(k) then k = self:FindBodygroupByName(k) end
+				self:SetBodygroup(k, v)
+			end
+		end
+
 		self:PhysicsInit(SOLID_VPHYSICS)
 		self:SetSolid(SOLID_VPHYSICS)
+		self:SetModel(model)
```

Comment:
You are setting the model twice

Reply:
Three times actually

Reply:
![image](https://user-images.githubusercontent.com/21306782/107873713-a36e1000-6eb4-11eb-94ce-e993837c1dfc.png)

Reply:
Good catch, fixed